src/data_conn.c

```diff
@@ -541,32 +545,33 @@ replica_thread(void *arg)
 	if (r_epollfd < 0) {
 		REPLICA_ERRLOG("epoll_create error for replica(%s:%d) "
 		    "errno(%d)\n", r->ip, r->port, errno);
-		return NULL;
+		goto initialize_error;
```

Comment:
this is good find..
let us do one more change:
- don't update 'replica' member variables till iofd is adding to epoll
- update 'replica' variables in the lock r_mtx when we are adding to epoll

Reply:
ok